src/HeaderCell.tsx

```diff
@@ -88,14 +87,47 @@ export default function HeaderCell<R, SR>({
     window.addEventListener('pointermove', onPointerMove);
     window.addEventListener('pointerup', onPointerUp);
   }
+  const index = sortColumns?.findIndex((sort) => sort.columnKey === column.key);
+  const sortDirection =
+    index !== undefined && index > -1 ? sortColumns?.[index].direction : undefined;
+  const priority = index !== undefined ? index + 1 : undefined;
+  const multiSorts = sortColumns ? sortColumns.length > 1 : false;
+  const sorted = priority ? priority > 0 : false;
+  const { sortDescendingFirst } = column;
+
+  const onSort = (columnKey: string, direction: SortDirection, ctrlClick: boolean) => {
+    if (!onSortColumnsChange) return;
+    if (ctrlClick) {
+      const newSorts = sortColumns ? [...sortColumns] : [];
+      if (index !== undefined) {
+        if (index > -1) {
+          if (sortDescendingFirst) {
+            if (sortDirection === 'DESC') newSorts[index] = { columnKey, direction };
+            else newSorts.splice(index, 1);
+          } else if (sortDirection === 'ASC') newSorts[index] = { columnKey, direction };
+          else newSorts.splice(index, 1);
+        } else {
+          newSorts.push({ columnKey, direction });
+        }
+
+        onSortColumnsChange([...newSorts]);
+      }
+    } else if (sortDirection === 'DESC') {
+      if (sortDescendingFirst) onSortColumnsChange([{ columnKey, direction }]);
+      else onSortColumnsChange([]);
+    } else if (sortDirection === 'ASC') {
+      if (sortDescendingFirst) onSortColumnsChange([]);
+      else onSortColumnsChange([{ columnKey, direction }]);
+    } else onSortColumnsChange([{ columnKey, direction }]);
+  };
```

Comment:
WDYT? 🤔 
```suggestion

  const sortIndex = sortColumns?.findIndex((sort) => sort.columnKey === column.key);
  const sortColumn =
    sortIndex !== undefined && sortIndex > -1 ? sortColumns![sortIndex] : undefined;
  const sortDirection = sortColumn?.direction;
  const priority = sortColumn !== undefined && sortColumns!.length > 1 ? sortIndex! + 1 : undefined;
  const ariaSort =
    sortDirection && !priority ? (sortDirection === 'ASC' ? 'ascending' : 'descending') : undefined;

  function onSort(ctrlClick: boolean) {
    if (onSortColumnsChange == null) return;

    const { sortDescendingFirst } = column;

    if (sortColumn === undefined) {
      // not currently sorted
      const nextSort: SortColumn = {
        columnKey: column.key,
        direction: sortDescendingFirst ? 'DESC' : 'ASC'
      };
      onSortColumnsChange(sortColumns && ctrlClick ? [...sortColumns, nextSort] : [nextSort]);
    } else {
      const nextSortColumns = [...sortColumns!];
      if (
        (sortDescendingFirst && sortDirection === 'ASC') ||
        (!sortDescendingFirst && sortDirection === 'DESC')
      ) {
        // remove sort
        nextSortColumns.splice(sortIndex!, 1);
      } else {
        // swap direction
        nextSortColumns[sortIndex!] = {
          columnKey: column.key,
          direction: sortDirection === 'ASC' ? 'DESC' : 'ASC'
        };
      }
      onSortColumnsChange(nextSortColumns);
    }
  }
```

Reply:
In `onSort`:
before: 11 `if`s + 9 `else`s
after: 3 `if`s + 2 `else`s
I'd say that's an improvement 🤔